cli/tools/run/hot_reload/mod.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+use std::{collections::HashMap, path::PathBuf};
+
+use deno_core::{
+  error::AnyError,
+  futures::StreamExt,
+  serde_json::{self, json},
+  LocalInspectorSession,
+};
+use deno_runtime::colors;
+use tokio::select;
+
+use crate::tools::run::hot_reload::json_types::{
+  RpcNotification, ScriptParsed, SetScriptSourceReturnObject, Status,
+};
+
+mod json_types;
+
+pub struct HotReloadManager {
+  session: LocalInspectorSession,
+  path_change_receiver: tokio::sync::broadcast::Receiver<Vec<PathBuf>>,
+  script_ids: HashMap<String, String>,
+}
+
+impl HotReloadManager {
+  pub fn new(
+    session: LocalInspectorSession,
+    path_change_receiver: tokio::sync::broadcast::Receiver<Vec<PathBuf>>,
+  ) -> Self {
+    Self {
+      session,
+      path_change_receiver,
+      script_ids: HashMap::new(),
+    }
+  }
+
+  // FIXME(SyrupThinker): Inspector code duplication
+
+  pub async fn start(&mut self) -> Result<(), AnyError> {
+    self.enable_debugger().await?;
```

Comment:
This is duplicated in `cli/tools/coverage.rs` - I think I will abstract it away to a helper struct that wraps `LocalInspectorSession` and provides some methods to interact with the debugger